core/src/main/java/org/keycloak/sdjwt/SdJwtVerificationContext.java

```diff
@@ -0,0 +1,730 @@
+        );
+
+        // Ensure that a signing algorithm was used that was deemed secure for the application.
+        // The none algorithm MUST NOT be accepted.
+        var holderVerifier = buildHolderVerifier(cnf);
+
+        // Validate the signature over the Key Binding JWT
+        try {
+            keyBindingJwt.verifySignature(holderVerifier);
+        } catch (VerificationException e) {
+            throw new VerificationException("Key binding JWT invalid", e);
+        }
+
+        // Check that the creation time of the Key Binding JWT is within an acceptable window.
+        validateKeyBindingJwtTimeClaims(keyBindingJwtVerificationOpts);
+
+        // Determine that the Key Binding JWT is bound to the current transaction and was created
+        // for this Verifier (replay protection) by validating nonce and aud claims.
+        preventKeyBindingJwtReplay(keyBindingJwtVerificationOpts);
+
+        // The same hash algorithm as for the Disclosures MUST be used (defined by the _sd_alg element
+        // in the Issuer-signed JWT or the default value, as defined in Section 5.1.1).
+        validateKeyBindingJwtSdHashIntegrity();
+
+        // Check that the Key Binding JWT is a valid JWT in all other respects
+        // -> Covered in part by `keyBindingJwt` being an instance of SdJws?
+        // -> Time claims are checked above
+    }
+
+    /**
+     * Validate Key Binding JWT's typ header attribute
+     *
+     * @throws VerificationException if verification failed
+     */
+    private void validateKeyBindingJwtTyp() throws VerificationException {
+        var typ = keyBindingJwt.getHeader().getType();
+        if (!typ.equals(KeyBindingJWT.TYP)) {
+            throw new VerificationException("Key Binding JWT is not of declared typ " + KeyBindingJWT.TYP);
+        }
+    }
+
+    /**
+     * Build holder verifier from JWK node.
+     *
+     * @throws VerificationException if unable
+     */
+    private SignatureVerifierContext buildHolderVerifier(JsonNode cnf) throws VerificationException {
+        Objects.requireNonNull(cnf);
+
+        // Read JWK
+        var cnfJwk = cnf.get("jwk");
+        if (cnfJwk == null) {
+            throw new UnsupportedOperationException("Only cnf/jwk claim supported");
+        }
+
+        // Parse JWK
+        KeyWrapper keyWrapper;
+        try {
+            JWK jwk = SdJwtUtils.mapper.convertValue(cnfJwk, JWK.class);
+            keyWrapper = JWKSUtils.getKeyWrapper(jwk);
+            Objects.requireNonNull(keyWrapper);
+        } catch (Exception e) {
+            throw new VerificationException("Malformed or unsupported cnf/jwk claim");
+        }
+
+        // Build verifier
+
+        // KeyType.EC
+        if (keyWrapper.getType().equals(KeyType.EC)) {
+            if (keyWrapper.getAlgorithm() == null) {
+                Objects.requireNonNull(keyWrapper.getCurve());
+
+                String alg = null;
+                switch (keyWrapper.getCurve()) {
```

Comment:
This switch clause can be represented simply as follows.
```
                String alg = switch (keyWrapper.getCurve()) {
                    case "P-256" -> "ES256";
                    case "P-384" -> "ES384";
                    case "P-521" -> "ES512";
                    default -> null;
                };
```

Reply:
Thank you for recommending this simpler syntax. Actually, my IDE has also been prompting for this. However, as the `core` module in locked in Java 11, we cannot use switch expressions.
https://github.com/keycloak/keycloak/blob/adb2af442a88b88452cd8aee2d064c906d4eb9ec/core/pom.xml#L34-L39

Reply:
Thank you for the clarification.